CONTRIBUTING.md

```diff
@@ -576,6 +575,60 @@ that are part of this project but not production code. The canonical example
 of this is `junit`.</dd>
 </dl>
 
+
+Reviewing and accepting your contribution
+-----------------------------------------
+
+We review every contribution carefully to ensure that the change is of high
+quality and fits well with the rest of the Elasticsearch codebase. If accepted,
+we will merge your change and usually take care of backporting it to
+appropriate branches ourselves.
+
+We really appreciate everyone who is interested in contributing to
+Elasticsearch and regret that we sometimes have to reject contributions even
+when they might appear to make genuine improvements to the system. Reviewing
+contributions can be a very time-consuming task, yet the review team is small
+and our time is very limited. In some cases the time we would need to spend on
+reviews would outweigh the benefits of a change by a significant margin.
+
+We strongly recommend you discuss your change in a Github issue before spending
```

Comment:
In my experience we don't always give a ton of guidance on "good first issues", but if folks come and ask we will almost certainly help. I think is important to check in even on "good first issues" just to claim them.

I think some of your concern here is around folks proposing a new feature that don't really understand what Elasticsearch *is* at all the right levels. Too much talking is never bad, but I think somehow this can be combined with pointing folks to "good first issue".